### ISSUE TITLE
[Proposal] IsNull / IsNotNull Converters

### ISSUE THREAD

**Issue by IeuanWalker** [CommunityToolkit/Maui#306]
# IsNull IsNotNull Converters

* [x] Proposed
* [x] Prototype
* [x] Implementation
  * [x] iOS Support
  * [x] Android Support
  * [x] macOS Support
  * [x] Windows Support
* [x] Unit Tests
* [x] Sample
* [x] Documentation: Done

## Link to Discussion

[Discussion](https://github.com/CommunityToolkit/Maui/discussions/305)

## Summary
[summary]: #summary

The `IsNullConverter` is a converter that allows users to convert an incoming binding to a bool value. This value represents if the incoming binding value is null.

The `IsNotNullConverter` is a converter that allows users to convert an incoming binding to a bool value. This value represents if the incoming binding value is not null.

## Motivation
[motivation]: #motivation

As mentioned in the discussion this was the behaviour of another converter, so these converters would just reimplement the old behaviour. It's been quite common in the projects I've worked on to be able to check any object type if it's null or not. 

## Detailed Design
[design]: #detailed-design

IsNotNullConverter.shared.cs
```csharp
/// <summary>
/// Converts the incoming value to a <see cref="bool"/> indicating whether or not the value is not null.
/// </summary>
public class IsNotNullConverter : BaseConverter<object?, bool>
{
	/// <summary>
	/// Converts the incoming object to a <see cref="bool"/> indicating whether or not the value is not null.
	/// </summary>
	/// <param name="value">The value to convert.</param>
	/// <returns>A <see cref="bool"/> indicating if the incoming value is not null</returns>
	public bool ConvertFrom(object? value) =>  value is not null;
}
```

IsNullConverter.shared.cs
```csharp
/// <summary>
/// Converts the incoming value to a <see cref="bool"/> indicating whether or not the value is null.
/// </summary>
public class IsNullConverter : BaseConverter<object?, bool>
{
	/// <summary>
	/// Converts the incoming object to a <see cref="bool"/> indicating whether or not the value is not null.
	/// </summary>
	/// <param name="value">The value to convert.</param>
	/// <returns>A <see cref="bool"/> indicating if the incoming value is not null</returns>
	public bool ConvertFrom(object? value) =>  value is null;
}
```

## Usage Syntax
[usage]: #usage-syntax

### XAML Usage

```xaml
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:xct="http://xamarin.com/schemas/2020/toolkit"
             x:Class="MyLittleApp.MainPage">

    <ContentPage.Resources>
        <ResourceDictionary>
            <xct: IsNullConverter x:Key="IsNullConverter" />
        </ResourceDictionary>
    </ContentPage.Resources>

    <StackLayout IsVisible="{Binding MyObject, Converter={StaticResource IsNullConverter}}">

    </StackLayout>
</ContentPage>
```

### C# Usage

```csharp
class MyPage : ContentPage
{
  public MyPage()
  {
    Content = new StackLayout().Bind(StackLayout.IsVisibleProperty, nameof(ViewModel.MyObject), converter: new IsNullConverter())
  }
}
```

## Drawbacks
[drawbacks]: #drawbacks

None

## Alternatives
[alternatives]: #alternatives

None

**Comment by IeuanWalker** [comment]
If approved, I'm happy to put a PR together for it

**Comment by brminnick** [comment]
Thanks @IeuanWalker! This all looks good!

The only thing I'd tweak is to use `is null` and `is not null` just to match our [Contribution Guidelines](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md). 

I'll raise this for a vote/approval at our next standup, 03 March.

**Comment by VladislavAntonyuk** [comment]
can we use `BaseConverter<object?, bool>`? example: https://github.com/CommunityToolkit/Maui/pull/327/files
I approve

**Comment by bijington** [comment]
I vote to approve

**Comment by IeuanWalker** [comment]
@brminnick awesome, ye will do

**Comment by IeuanWalker** [comment]
@brminnick I can't seem to get the sample project to run. These are the only options showing - 
![image](https://user-images.githubusercontent.com/6544051/158651030-2349ea9d-206c-43c4-94c7-c9dde3e7d01a.png)
And get this message if I try and run anyway -
![image](https://user-images.githubusercontent.com/6544051/158651092-49b89618-05be-446a-8e24-fbb75a316e9c.png)


I have my own project using maui that's working fine - 
![image](https://user-images.githubusercontent.com/6544051/158651252-39dc85d7-4d90-4c09-8c4e-62c77d80a81b.png)

**Comment by VladislavAntonyuk** [comment]
open csproj and check that `<OutputType>Exe</OutputType>`

**Comment by IeuanWalker** [comment]
@VladislavAntonyuk ye thats in the csproj.

How do i run it with CLI?

Im getting these errors now too, appeared after i cleaned the solution, restarted VS and built the solution -
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0534	'PopupHandler' does not implement inherited abstract member 'ElementHandler<IPopup, object>.CreatePlatformElement()'	CommunityToolkit.Maui.Core (net6.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.shared.cs	6	Active
Error	CS0115	'PopupHandler.CreateNativeElement()': no suitable method found to override	CommunityToolkit.Maui.Core (net6.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.shared.cs	56	Active
Error	CS0534	'PopupHandler' does not implement inherited abstract member 'ElementHandler<IPopup, MauiPopup>.CreatePlatformElement()'	CommunityToolkit.Maui.Core (net6.0-android), CommunityToolkit.Maui.Core (net6.0-maccatalyst), CommunityToolkit.Maui.Core (net6.0-windows10.0.19041.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.shared.cs	6	Active
Error	CS0534	'PopupHandler' does not implement inherited abstract member 'ElementHandler<IPopup, MauiPopup>.CreatePlatformElement()'	CommunityToolkit.Maui.Core (net6.0-ios)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	6	Active
Error	CS0115	'PopupHandler.CreateNativeElement()': no suitable method found to override	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	95	Active
Error	CS0534	'PopupHandler' does not implement inherited abstract member 'ElementHandler<IPopup, MauiPopup>.CreatePlatformElement()'	CommunityToolkit.Maui.Core (net6.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopUpHandler.android.cs	7	Active
Error	CS0115	'PopupHandler.CreateNativeElement()': no suitable method found to override	CommunityToolkit.Maui.Core (net6.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopUpHandler.android.cs	98	Active
Error	CS0534	'PopupHandler' does not implement inherited abstract member 'ElementHandler<IPopup, MauiPopup>.CreatePlatformElement()'	CommunityToolkit.Maui.Core (net6.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	6	Active
Error	CS0115	'PopupHandler.CreateNativeElement()': no suitable method found to override	CommunityToolkit.Maui.Core (net6.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	95	Active
Error	CS0115	'PopupHandler.CreateNativeElement()': no suitable method found to override	CommunityToolkit.Maui.Core (net6.0)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopUpHandler.windows.cs	90	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	16	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	22	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	33	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	35	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	52	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	53	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	63	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	73	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	83	Active
Error	CS1061	'PopupHandler' does not contain a definition for 'NativeView' and no accessible extension method 'NativeView' accepting a first argument of type 'PopupHandler' could be found (are you missing a using directive or an assembly reference?)	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	84	Active
Error	CS0103	The name 'NativeView' does not exist in the current context	CommunityToolkit.Maui.Core (net6.0-ios), CommunityToolkit.Maui.Core (net6.0-maccatalyst)	D:\WS\GitHub\MauiToolKit\src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs	104	Active

```

**Comment by pictos** [comment]
@IeuanWalker did you update your MAUI to P14?

**Comment by VladislavAntonyuk** [comment]
yes, it is MAUI Preview 14. The PR with the update is under review.

**Comment by IeuanWalker** [comment]
Cool no prob, i'll wait

**Comment by brminnick** [comment]
@IeuanWalker You should be unblocked now that we merged in support for Preview 14: #325 🙌
